In [1]:
import os, json, math, time
# Standard Imports
import pandas as pd
from tqdm.notebook import tqdm_notebook
FOLDER = "Data/"
os.makedirs(FOLDER, exist_ok=True)
os.listdir(FOLDER)

['.ipynb_checkpoints',
 'data_enrichment_belt.json',
 'final_tmdb_data_2000.csv.gz',
 'final_tmdb_data_2001.csv.gz',
 'medical_data.csv',
 'superhero_info.csv',
 'superhero_powers - superhero_powers.csv',
 'title_akas.csv.gz',
 'title_basics.csv.gz',
 'title_title.csv.gz',
 'tmbd_api_results2000.json',
 'tmbd_api_results2001.json',
 'tmdb_api_results_.json',
 'tmdb_results_combined.csv.gz']

In [2]:
import json
with open("/Users/Meryum Syeda/.secret/tmdb_api.json", "r") as f: #change the path to match YOUR path!!
    login = json.load(f)
login.keys()

dict_keys(['api-key'])

In [3]:
import tmdbsimple as tmdb
tmdb.API_KEY =  login['api-key']

In [4]:
## make a movie object using the .Movies function from tmdb
movie = tmdb.Movies(603)

In [5]:
## movie objects have a .info dictionary 
info = movie.info()
info

{'adult': False,
 'backdrop_path': '/n2nm4aZRmXyJ9LT4xQX9X6ThcP7.jpg',
 'belongs_to_collection': {'id': 2344,
  'name': 'The Matrix Collection',
  'poster_path': '/bV9qTVHTVf0gkW0j7p7M0ILD4pG.jpg',
  'backdrop_path': '/bRm2DEgUiYciDw3myHuYFInD7la.jpg'},
 'budget': 63000000,
 'genres': [{'id': 28, 'name': 'Action'},
  {'id': 878, 'name': 'Science Fiction'}],
 'homepage': 'http://www.warnerbros.com/matrix',
 'id': 603,
 'imdb_id': 'tt0133093',
 'original_language': 'en',
 'original_title': 'The Matrix',
 'overview': 'Set in the 22nd century, The Matrix tells the story of a computer hacker who joins a group of underground insurgents fighting the vast and powerful computers who now rule the earth.',
 'popularity': 68.99,
 'poster_path': '/f89U3ADr1oiB1s9GkdPOEpXUk5H.jpg',
 'production_companies': [{'id': 79,
   'logo_path': '/tpFpsqbleCzEE2p5EgvUq6ozfCA.png',
   'name': 'Village Roadshow Pictures',
   'origin_country': 'US'},
  {'id': 174,
   'logo_path': '/IuAlhI9eVC9Z8UQWOIDdWRKSEJ.png',

In [6]:
info['revenue']

463517383

In [7]:
info['imdb_id']

'tt0133093'

In [8]:
info['budget']

63000000

In [9]:
YEARS_TO_GET = [2000,2001]

In [10]:
# Function to get movie ratings
def get_movie_with_rating(movie_id):
    movie = tmdb.Movies(movie_id)
    movie_info = movie.info()
    releases = movie.releases()
    for c in releases['countries']:
        if c['iso_3166_1'] == 'US':
             movie_info['certification'] = c['certification']
    return movie_info

In [11]:
def write_json(new_data, filename): 
    
    with open(filename,'r+') as file:
        # First we load existing data into a dict.
        file_data = json.load(file)
        ## Choose extend or append
        if (type(new_data) == list) & (type(file_data) == list):
            file_data.extend(new_data)
        else:
             file_data.append(new_data)
        # Sets file's current position at offset.
        file.seek(0)
        # convert back to json.
        json.dump(file_data, file)
    
    # If it does not exist: create it
        if file_exists == False:
    # save an empty dict with just "imdb_id" to the new json file.
           with open(JSON_FILE,'w') as f:
                json.dump([{'imdb_id':0}],f)

In [12]:
for YEAR in tqdm_notebook(YEARS_TO_GET,desc='YEARS',position=0):
    
    JSON_FILE = f'{FOLDER}tmbd_api_results{YEAR}.json'
    # Check if file exists
    file_exists = os.path.isfile(JSON_FILE)
# If it does not exist: create it
    if file_exists == False:
# save an empty dict with just "imdb_id" to the new json file.
        with open(JSON_FILE,'w') as f:
            json.dump([{'imdb_id':0}],f)
    
    basics = pd.read_csv("Data/title_basics.csv.gz", low_memory = False)
    
    # Filter the IMDB title_basics file on current Year
    df = basics.loc[ basics['startYear']==YEAR].copy()
    
    # Save IMDB title_basics file movie ids to list
    movie_ids = df['tconst'].copy()
    
    # Load existing data from json into a dataframe called "previous_df"
    previous_df = pd.read_json(JSON_FILE)

    # filter out any ids that are already in the JSON_FILE
    movie_ids_to_get = movie_ids[~movie_ids.isin(previous_df['imdb_id'])]

#Get index and movie id from list


# INNER Loop
    for movie_id in tqdm_notebook(movie_ids_to_get,
                                  desc=f'Movies from {YEAR}',
                                  position=1,
                                  leave=True):
    
# Attempt to retrieve then data for the movie id
        try:
            temp = get_movie_with_rating(movie_id)  #This uses your pre-made function
            # Append/extend results to existing file using a pre-made function
            write_json(temp,JSON_FILE)
            # Short 20 ms sleep to prevent overwhelming server
            time.sleep(0.02)
            
        # If it fails,  make a dict with just the id and None for certification.
        except Exception as e:
            continue
            
   #Saving new year as the current df
    #df = basics.loc[ basics['startYear']==YEAR].copy()
    final_year_df = pd.read_json(JSON_FILE)
    final_year_df.to_csv(f"{FOLDER}final_tmdb_data_{YEAR}.csv.gz", compression="gzip", index=False)

YEARS:   0%|          | 0/2 [00:00<?, ?it/s]

Movies from 2000:   0%|          | 0/235 [00:00<?, ?it/s]

Movies from 2001:   0%|          | 0/265 [00:00<?, ?it/s]

In [13]:
df_1 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz")
df_2 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz")

In [25]:
df_1.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         0.000        0.0           NaN  
4    0.0         8.106     1957.0            PG  

[5 rows x 26 columns]

In [26]:
df_2.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /ab5yL8zgRotrICzGbEl10z24N71.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118154    0.0                               NaN                   NaN   
4  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 14, 'nam...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3         0.0                [{'id': 99, 'name': 'Documentary'}]      NaN   
4  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   

         id original_language    original_title  ...     revenue  runtime  \
0       NaN               NaN               NaN  ...         NaN      NaN   
1   11232.0                en    Kate & Leopold  ...  76019048.0    118.0   
2  151007.0                en  The Silent Force  ...         0.0     90.0   
3  292980.0                en       White Hotel  ...         0.0     90.0   
4   10696.0                en           Glitter  ...   5271666.0    104.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Amharic', 'iso_639_1': 'am'...  Released   
4  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline             title  video  \
0                                                NaN               NaN    NaN   
1  If they lived in the same century they'd be pe...    Kate & Leopold    0.0   
2  They left him for dead... They should have fin...  The Silent Force    0.0   
3                                                NaN       White Hotel    0.0   
4   In music she found her dream, her love, herself.           Glitter    0.0   

   vote_average vote_count certification  
0           NaN        NaN           NaN  
1           6.3     1137.0         PG-13  
2           5.0        3.0           NaN  
3           0.0        0.0           NaN  
4           4.6      117.0         PG-13  

[5 rows x 26 columns]

In [28]:
df_1.shape

(1275, 26)

In [29]:
df_2.shape

(1344, 26)

In [14]:
df = df_1.append(df_2)

C:\Users\Meryum Syeda\AppData\Local\Temp\ipykernel_836\2979329560.py:1: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df_1.append(df_2)


In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2619 entries, 0 to 1343
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2619 non-null   object 
 1   adult                  2617 non-null   float64
 2   backdrop_path          1327 non-null   object 
 3   belongs_to_collection  197 non-null    object 
 4   budget                 2617 non-null   float64
 5   genres                 2617 non-null   object 
 6   homepage               189 non-null    object 
 7   id                     2617 non-null   float64
 8   original_language      2617 non-null   object 
 9   original_title         2617 non-null   object 
 10  overview               2563 non-null   object 
 11  popularity             2617 non-null   float64
 12  poster_path            2291 non-null   object 
 13  production_companies   2617 non-null   object 
 14  production_countries   2617 non-null   object 
 15  rele

In [16]:
budget_filter = df['budget']>0
budget_filter.head()
df = df.loc[budget_filter, :]
df.head()

imdb_id  adult                     backdrop_path  \
1   tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg   
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
6   tt0119273    0.0  /fClJrAmJQ90zg9gWnfHVYSMiDTv.jpg   
11  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
12  tt0120626    0.0  /6xz9H7IMK3W21SNwAVmE9oW6YjX.jpg   

                                belongs_to_collection      budget  \
1                                                 NaN  10000000.0   
4                                                 NaN    150000.0   
6   {'id': 141086, 'name': 'Heavy Metal Collection...  15000000.0   
11                                                NaN    120000.0   
12                                                NaN   6105121.0   

                                               genres  \
1   [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
6   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
12  [{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...   

                                homepage       id original_language  \
1                                    NaN  62127.0                en   
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
6                                    NaN  16225.0                en   
11                                   NaN  19085.0                en   
12                                   NaN  54380.0                pt   

       original_title  ...     revenue  runtime  \
1     The Fantasticks  ...         0.0     86.0   
4                花樣年華  ...  12854953.0     99.0   
6    Heavy Metal 2000  ...         0.0     88.0   
11             Vulgar  ...     14904.0     87.0   
12  Capitães de Abril  ...         0.0    123.0   

                                     spoken_languages    status  \
1   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
6   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
12  [{'english_name': 'Portuguese', 'iso_639_1': '...  Released   

                                              tagline                 title  \
1       Try to remember the first time magic happened       The Fantasticks   
4   Feel the heat, keep the feeling burning, let t...  In the Mood for Love   
6   not to survive the fight in the external world...      Heavy Metal 2000   
11   Everyone loves a clown... some more than others.                Vulgar   
12                                                NaN     Captains of April   

    video  vote_average vote_count certification  
1     0.0         5.500       22.0           NaN  
4     0.0         8.106     1957.0            PG  
6     0.0         6.086      157.0             R  
11    0.0         5.400       42.0             R  
12    0.0         6.400       47.0           NaN  

[5 rows x 26 columns]

In [17]:
revenue_filter = df['revenue']>0
revenue_filter.head()
df = df.loc[revenue_filter, :]
df.head()

imdb_id  adult                     backdrop_path  \
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
11  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
13  tt0120630    0.0  /sPAwM8WxMdXNlqeP4F9DMe9LYoY.jpg   
16  tt0120753    0.0                               NaN   
17  tt0120755    0.0  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                                belongs_to_collection       budget  \
4                                                 NaN     150000.0   
11                                                NaN     120000.0   
13  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
16                                                NaN    8000000.0   
17  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                               genres  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...   
17  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                homepage       id original_language  \
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
11                                   NaN  19085.0                en   
13                                   NaN   7443.0                en   
16                                   NaN    318.0                en   
17                                   NaN    955.0                en   

              original_title  ...      revenue  runtime  \
4                       花樣年華  ...   12854953.0     99.0   
11                    Vulgar  ...      14904.0     87.0   
13               Chicken Run  ...  224834564.0     84.0   
16  The Million Dollar Hotel  ...     105983.0    122.0   
17    Mission: Impossible II  ...  546388105.0    123.0   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   
17                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
4       In the Mood for Love    0.0         8.106     1957.0            PG  
11                    Vulgar    0.0         5.400       42.0             R  
13               Chicken Run    0.0         6.726     4109.0             G  
16  The Million Dollar Hotel    0.0         5.915      254.0             R  
17    Mission: Impossible II    0.0         6.109     5448.0         PG-13  

[5 rows x 26 columns]

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 351 entries, 4 to 1299
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                351 non-null    object 
 1   adult                  351 non-null    float64
 2   backdrop_path          338 non-null    object 
 3   belongs_to_collection  84 non-null     object 
 4   budget                 351 non-null    float64
 5   genres                 351 non-null    object 
 6   homepage               57 non-null     object 
 7   id                     351 non-null    float64
 8   original_language      351 non-null    object 
 9   original_title         351 non-null    object 
 10  overview               351 non-null    object 
 11  popularity             351 non-null    float64
 12  poster_path            350 non-null    object 
 13  production_companies   351 non-null    object 
 14  production_countries   351 non-null    object 
 15  relea

Below we can see there are 140 R-rated movies included, 117 PG-13, 29 PG, 13 G-rated, and 6 that do not have a rating.

In [19]:
df['certification'].value_counts()

R        140
PG-13    117
PG        29
G         13
NR         6
Name: certification, dtype: int64

In [20]:
df["certification"]

4          PG
11          R
13          G
16          R
17      PG-13
        ...  
1027      NaN
1059      NaN
1167      NaN
1280      NaN
1299      NaN
Name: certification, Length: 351, dtype: object

In [21]:
#Average revenue per certification category
df.groupby('certification')['revenue'].mean()

certification
G        1.332169e+08
NR       2.382332e+07
PG       1.346138e+08
PG-13    1.107121e+08
R        5.344654e+07
Name: revenue, dtype: float64

In [22]:
#Average budeget per certification category
df.groupby('certification')['budget'].mean()

certification
G        4.400000e+07
NR       1.478333e+07
PG       5.374138e+07
PG-13    4.680682e+07
R        2.750255e+07
Name: budget, dtype: float64

In [23]:
df.head()

imdb_id  adult                     backdrop_path  \
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
11  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
13  tt0120630    0.0  /sPAwM8WxMdXNlqeP4F9DMe9LYoY.jpg   
16  tt0120753    0.0                               NaN   
17  tt0120755    0.0  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                                belongs_to_collection       budget  \
4                                                 NaN     150000.0   
11                                                NaN     120000.0   
13  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
16                                                NaN    8000000.0   
17  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                               genres  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...   
17  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                                homepage       id original_language  \
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
11                                   NaN  19085.0                en   
13                                   NaN   7443.0                en   
16                                   NaN    318.0                en   
17                                   NaN    955.0                en   

              original_title  ...      revenue  runtime  \
4                       花樣年華  ...   12854953.0     99.0   
11                    Vulgar  ...      14904.0     87.0   
13               Chicken Run  ...  224834564.0     84.0   
16  The Million Dollar Hotel  ...     105983.0    122.0   
17    Mission: Impossible II  ...  546388105.0    123.0   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   
17                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
4       In the Mood for Love    0.0         8.106     1957.0            PG  
11                    Vulgar    0.0         5.400       42.0             R  
13               Chicken Run    0.0         6.726     4109.0             G  
16  The Million Dollar Hotel    0.0         5.915      254.0             R  
17    Mission: Impossible II    0.0         6.109     5448.0         PG-13  

[5 rows x 26 columns]

In [24]:
## Save current dataframe to file.
df.to_csv("Data/tmdb_results_combined.csv.gz",compression='gzip',index=False)
# Open saved file and preview again
df = pd.read_csv("Data/tmdb_results_combined.csv.gz", low_memory = False)
df.head()

imdb_id  adult                     backdrop_path  \
0  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
1  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
2  tt0120630    0.0  /sPAwM8WxMdXNlqeP4F9DMe9LYoY.jpg   
3  tt0120753    0.0                               NaN   
4  tt0120755    0.0  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                               belongs_to_collection       budget  \
0                                                NaN     150000.0   
1                                                NaN     120000.0   
2  {'id': 718551, 'name': 'Chicken Run Collection...   45000000.0   
3                                                NaN    8000000.0   
4  {'id': 87359, 'name': 'Mission: Impossible Col...  125000000.0   

                                              genres  \
0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
1  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   
2  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
3  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...   
4  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   

                               homepage       id original_language  \
0  http://www.wkw-inthemoodforlove.com/    843.0                cn   
1                                   NaN  19085.0                en   
2                                   NaN   7443.0                en   
3                                   NaN    318.0                en   
4                                   NaN    955.0                en   

             original_title  ...      revenue  runtime  \
0                      花樣年華  ...   12854953.0     99.0   
1                    Vulgar  ...      14904.0     87.0   
2               Chicken Run  ...  224834564.0     84.0   
3  The Million Dollar Hotel  ...     105983.0    122.0   
4    Mission: Impossible II  ...  546388105.0    123.0   

                                    spoken_languages    status  \
0  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline  \
0  Feel the heat, keep the feeling burning, let t...   
1   Everyone loves a clown... some more than others.   
2  This ain't no chick flick. It's poultry in mot...   
3                    Everyone has something to hide.   
4                       Expect the impossible again.   

                      title  video  vote_average vote_count certification  
0      In the Mood for Love    0.0         8.106     1957.0            PG  
1                    Vulgar    0.0         5.400       42.0             R  
2               Chicken Run    0.0         6.726     4109.0             G  
3  The Million Dollar Hotel    0.0         5.915      254.0             R  
4    Mission: Impossible II    0.0         6.109     5448.0         PG-13  

[5 rows x 26 columns]